cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -179,9 +188,18 @@ func (k *proxy) newClientSet() (*kubernetes.Clientset, error) {
 		return nil, err
 	}
 
-	cs, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to create the client-go client")
+	var cs *kubernetes.Clientset
+	// Nb. The operation is wrapped in a retry loop to make newClientSet more resilient to temporary connection problems.
+	connectBackoff := newConnectBackoff()
+	if err := retryWithExponentialBackoff(connectBackoff, func() error {
+		var err error
+		cs, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			return errors.Wrap(err, "failed to create the client-go client")
```

Comment:
do we need to wrap here, or should we wrap on line 202 instead?

Reply:
done